real_time/surface_src/java_src/Alice/src/test/java/com/roboclub/robobuggy/main/RobobuggyConfigFileTest.java

```diff
@@ -0,0 +1,216 @@
+            e.printStackTrace();
+            Assert.fail();
+        }
+        assert RobobuggyConfigFile.getRobot() instanceof TransistorDataCollection;
+    }
+
+    /**
+     * Creates a config file that specifies a PlayBackRobot robot, then creates the robot
+     */
+    @Test
+    public void testPlayBackRobot(){
+        try {
+            createConfigFileWithRobot("PlayBackRobot");
+            RobobuggyConfigFile.loadConfigFile();
+        } catch (IOException e) {
+            e.printStackTrace();
+            Assert.fail();
+        } catch (IllegalArgumentException e) {
+            e.printStackTrace();
+            Assert.fail();
+        }
+        assert RobobuggyConfigFile.getRobot() instanceof PlayBackRobot;
+    }
+
+    /**
+     * Creates a config file that specifies a SimRobot robot, then creates the robot
+     */
+    @Test
+    public void testSimRobot(){
+        try {
+            createConfigFileWithRobot("SimRobot");
+            RobobuggyConfigFile.loadConfigFile();
+        } catch (IOException e) {
+            e.printStackTrace();
+            Assert.fail();
+        } catch (IllegalArgumentException e) {
+            e.printStackTrace();
+            Assert.fail();
+        }
+        assert RobobuggyConfigFile.getRobot() instanceof SimRobot;
+    }
+
+    /**
+     * Creates a config file with a robot field but no robot name
+     * It's expected to throw an IllegalArgumentException
+     */
+    @Test
+    public void testNoName(){
+        try {
+            createConfigFileWithRobot("");
+            RobobuggyConfigFile.loadConfigFile();
+            Assert.fail();
+        } catch (IOException e) {
+            e.printStackTrace();
+            Assert.fail();
+        } catch (IllegalArgumentException e) {
+            //Good;test passed
+        }
+    }
+
+    /**
+     * Creates a config file that specifies a robot but gives an invalid name
+     * It's expected to throw an IllegalArgumentException
+     */
+    @Test
+    public void testRandomName(){
+        try {
+            createConfigFileWithRobot("SafetyDuck!!!");
+            RobobuggyConfigFile.loadConfigFile();
+            Assert.fail();
+        } catch (IOException e) {
+            e.printStackTrace();
+            Assert.fail();
+        } catch (IllegalArgumentException e) {
+            //Good;test passed
+        }
+    }
+
+    /**
+     * Creates a config file with the robot field "NOT_SET"
+     * It's expected to throw an IllegalArgumentException
+     */
+    @Test
+    public void testNotSet(){
+        try {
+            createConfigFileWithRobot("NOT_SET");
+            RobobuggyConfigFile.loadConfigFile();
+            Assert.fail();
+        } catch (IOException e) {
+            e.printStackTrace();
+            Assert.fail();
+        } catch (IllegalArgumentException e) {
+            //Good;test passed
+        }
+    }
+
+    /**
+     * Creates a config file without a robot field
```

Comment:
(Creates a config file without a robot **key-value pair**) I think is a little clearer